components/SideNav.tsx

```diff
@@ -53,3 +40,47 @@ export default function SideNav({
         </>
     );
 }
+
+const SideNavItem = ({
+    item,
+    active,
+    setActive
+}: {
+    item: Documentation;
+    active: null | string;
+    setActive: React.Dispatch<null | string>;
```

Comment:
https://stackoverflow.com/questions/56028635/passing-usestate-as-props-in-typescript